lib/danger_plugin.rb

```diff
@@ -39,33 +39,67 @@ def lint_files
         return
       end
 
-      swiftlint_command = ""
-      swiftlint_command += "cd #{directory} && " if directory
-      swiftlint_command += "swiftlint lint --quiet --reporter json"
-      swiftlint_command += " --config #{config_file}" if config_file
-
-      require 'json'
-      result_json = JSON.parse(`(#{swiftlint_command})`).flatten
-
-      # Convert to swiftlint results
-      warnings = result_json.select do |results| 
-        results['severity'] == 'Warning'
-      end
-      errors = result_json.select do |results| 
-        results['severity'] == 'Error' 
-      end
-
-      message = ''
-
-      # We got some error reports back from swiftlint
-      if warnings.count > 0 || errors.count > 0
-        message = "### SwiftLint found issues\n\n"
+      require 'tempfile'
+      Tempfile.open('.swiftlint_danger.yml') do |temp_config_file|
+        cfp = nil
+        if config_file
+          require 'yaml'
+          original_config = YAML.load_file(config_file)
+          danger_compatible_config = original_config
+          danger_compatible_config['included'] = nil
+          temp_config_file.write danger_compatible_config.to_yaml
+
+          cfp = temp_config_file
+        end
+
+        swiftlint_command = "swiftlint lint --quiet --reporter json"
+        swiftlint_command += " --config #{cfp}" if cfp
+
+        require 'json'
+
+        if directory
+          swiftlint_command = "cd #{directory} && #{swiftlint_command}" if directory
+
+          result_json = JSON.parse(`(#{swiftlint_command})`).flatten
+        else
+          # Either use files provided, or use the modified + added
+          swift_files = files ? Dir.glob(files) : (git.modified_files + git.added_files)
+          swift_files.select! do |line| line.end_with?(".swift") end
+
+          # Make sure we don't fail when paths have spaces
+          swift_files = swift_files.map { |file| "\"#{file}\"" }
+
+          result_json = swift_files
+          .uniq
+          .collect { |f| JSON.parse(`(#{swiftlint_command} --path #{f})`.strip).flatten }
+          .flatten
+        end
+
+        # Convert to swiftlint results
+        warnings = result_json.select do |results| 
+          results['severity'] == 'Warning'
+        end
+        errors = result_json.select do |results| 
+          results['severity'] == 'Error' 
+        end
+
+        message = ''
+
+        # We got some error reports back from swiftlint
+        if warnings.count > 0 || errors.count > 0
+          message = "### SwiftLint found issues\n\n"
+        end
+
+        warnings.each do |warning|
+          location = warning['file'].gsub(Dir.pwd, '')
+          warn(warning['type'], file: location, line: warning['line'])
+        end
```

Comment:
As mentioned in the PR description, this was an experiment, might be better to have a dedicated PR for this.